Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:18.04
```

Comment:
It'll probably speed up our builds considerably and make them a bit more predictable if we moved to using a different docker base image, such as the official python3 base image. As a nice benefit, it'll let us simplify the logic here to something like 

```
FROM python:3
ADD . /selected-tests
WORKDIR /selected-tests 
RUN pip install -r requirements.txt
RUN pip install -e .

CMD ["scripts/server.sh"]
```

Reply:
See https://pythonspeed.com/articles/base-image-python-docker-images/

I don't think we should use just `python:3`, we should explicitly pick which distro to use. I have had problems with alpine in the past and would prefer not to use it.

Reply:
Also, just to note, every other project we have is using the ubuntu image. So using a different image could lead to this project behaving differently. If we used the python:3 debian image, they should be similar, but there still could be differences.

Reply:
I don't really see a risk in the distro being different across repos. Everything we're doing in them is just using the python commands and runtime so it's rather unlikely to be different across debian and ubuntu distros.

That's a fair point about alpine, it can be rather annoying at times. And we can pin it further to a specific minor or even patch version if people want.

```3.7.4-buster, 3.7-buster, 3-buster, buster```
```3.7.4-stretch, 3.7-stretch, 3-stretch, stretch```

All the above are available in docker hub. 

If moving to something besides ubuntu isn't favorable though, another option is to build an ubuntu python base image and push it to a registry and use that.

Reply:
Sounds like we want to stick to ubuntu for consistency, so I'm going to leave this as-is for now. I don't think there's a huge benefit from building a ubuntu python base image (given this Dockerfile is relatively small as-is) but am happy to discuss as a separate ticket.

Reply:
The main benefit to building an ubuntu image is the guarantee of the same runtime environment every time we build a new version off of that base. Currently, we're installing the same packages into a base ubuntu version every build which theoretically should give the same/a similar enough result but doesn't give a guarantee of it. 

The other big benefit here is build speed as rather than having to install the same packages across builds they'd already be installed into the base and all the logic in the dockerfile would have to do with how to run our code rather than a runtime. 

If the decision is to table it for now, that's fine.

Reply:
I think we table it for now.

We can have a separate discussion for what we want to do as a team and if we want switch over the Dockerfiles in all our projects to whatever we decide on.